ci/ray_ci/tester.py

```diff
@@ -13,6 +14,18 @@
 bazel_workspace_dir = os.environ.get("BUILD_WORKSPACE_DIRECTORY", "")
 
 
+class TestType(Enum):
+    ASAN = "asan_tests"
+    DATA_INTEGRATION = "data_integration"
+    DEBUG = "debug_tests"
+    DOC = "doctest"
+    GPU = "gpu"
+    POST_WHEEL = "post_wheel_build"
+    KUBERAY = "kuberay_operator"
+    SPARK_PLUGIN = "spark_plugin_tests"
+    WORKER_CONTAINER = "worker-container"
```

Comment:
I am not sure if this is a good move. I think it is not.

before this change, the "types" are just bazel test tags.
after this change, "types" become `ray_ci` builtin enum type, and adding or removing a "type" has to go through the ci team (a.k.a you..)

I am not really convinced of (or I just do not really understand) the benefit here.

Reply:
> so that people will stop, or get discouraged from, creating a new test types.

if that is the purpose. you can just add a lint to limit the tags? and add a note in that tag list like "please try to reuse tags if possible".

and I am also not sure people like it. what if they want new tags? want to create new types of tests? do you really want them to always come to you, explain to you and convince you that their need of new tag is justified? I do not think this is in (natural) devprod's scope. devprod has saying on "how to test", but not "what to test".

I am also not fully convinced why snow flake tags are bad.

maybe we should talk about the high level questions in a 1:1 ?

Reply:
totally, we can talk more about this maybe tomorrow ;)